### PR TITLE
chore(ci): add problem matcher for cppcheck

### PR DESCRIPTION
**Pull request by HansRobo**
## Description

Add problem matcher for cppcheck messages to annotate in file diff view in GitHub.

image

## How was this PR tested?

Check the cppcheck error is matched well.
https://github.com/autowarefoundation/autoware.universe/actions/runs/9763078681/job/26948068884?pr=7794#step:12:7

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: .github/cppcheck-problem-matcher.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "problemMatcher": [
+    {
+      "owner": "cppcheck",
+      "pattern": [
+        {
+          "regexp": "^([^:]+):(\\d+):(\\d*):\\s(style|performance|warning|error):\\s(.*)$",
```
Comment (veqcc) [inline .github/cppcheck-problem-matcher.json:7]: portability
を追加でお願いします！

File: .github/workflows/cppcheck-daily.yaml [hunk 2]
```diff
@@ -18,6 +18,10 @@ jobs:
         run: |
           sudo snap install cppcheck
 
+      - name: Setup problem matchers for cppcheck
+        run: echo "::add-matcher::.github/cppcheck-problem-matcher.json"
+
+      # cspell: ignore suppr
```
Comment (veqcc) [inline .github/workflows/cppcheck-daily.yaml:24]: How the problem matcher works for -daily workflow? How is it shown to users?

File: .github/cppcheck-problem-matcher.json (hunk 1 above)
Comment (HansRobo) [reply .github/cppcheck-problem-matcher.json:7]:
```suggestion
          "regexp": "^([^:]+):(\\d+):(\\d*):\\s(style|portability|performance|warning|error):\\s(.*)$",
```

File: .github/workflows/cppcheck-daily.yaml (hunk 2 above)
Comment (HansRobo) [reply .github/workflows/cppcheck-daily.yaml:24]: You don't have the file diff view for -dialy workflow.
But you can check matched errors in workflow summary view.
https://github.com/autowarefoundation/autoware.universe/actions/runs/9763078681?pr=7794
image

Comment (HansRobo) [reply .github/workflows/cppcheck-daily.yaml:24]: Testing in https://github.com/HansRobo/autoware.universe/actions/runs/9769570736

Comment (veqcc) [reply .github/workflows/cppcheck-daily.yaml:24]: Understood!

One concern is that cppcheck-daily ditects around 8,000 warnings.
Are those warnings all displayed in summary view?

Comment (HansRobo) [reply .github/workflows/cppcheck-daily.yaml:24]: Oh, sorry.
The output format in -dialy workflow is different from format in -differential workflow.
I found that it cannot be applied as it is, so I will delete it in -dialy workflow for now.

Comment (HansRobo) [reply .github/workflows/cppcheck-daily.yaml:24]:
```suggestion
      # cspell: ignore suppr
```

**Review by veqcc — APPROVED** [review]
LGTM :+1:

**Pull request merged**